### PR TITLE
Add fbclient/3.0.8

### PR DESCRIPTION
**Pull request by friendlyanon**
Specify library name and version:  **fbclient/3.0.8**

Issue: #8618

The included build scripts are unusable for the purposes of cleanly building the `fbclient` library. Upstream build scripts are also using version scripts to supposedly control symbol visibility on *nix systems, but I couldn't get it to work, so I just left all symbols visible on *nix platforms. Windows has a proper `.def` file, so that's good there.

`array.cpp` and `blob.cpp` are included in the recipe, because of https://github.com/conan-io/conan-center-index/issues/9307

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/fbclient/3.x.x/array.cpp [hunk 1]
```diff
@@ -0,0 +1,889 @@
+/*********** Preprocessed module -- do not edit ***************/
```
Comment (SSE4) [inline recipes/fbclient/3.x.x/array.cpp:1]: did you generate it somehow?

File: recipes/fbclient/3.x.x/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,280 @@
+cmake_minimum_required(VERSION 3.16)
```
Comment (SSE4) [inline recipes/fbclient/3.x.x/CMakeLists.txt:1]: is it taken from somewhere or written from scratch?

File: recipes/fbclient/3.x.x/array.cpp (hunk 1 above)
Comment (friendlyanon) [reply recipes/fbclient/3.x.x/array.cpp:1]: Refer ro OP and the mentioned issue.

File: recipes/fbclient/3.x.x/CMakeLists.txt (hunk 2 above)
Comment (friendlyanon) [reply recipes/fbclient/3.x.x/CMakeLists.txt:1]: Refer to OP. I wrote this.

**Comment by friendlyanon** [comment]
Based on feedback from Slack, I just set things back to CMake 3.1 minimum and the `-x objective-c` flags manually set for the lone `.m` file.

**Comment by uilianries** [comment]
@friendlyanon you commented about the need of both `array.cpp` and `blob.cpp`, but you didn't mention their origin. Did you wrote those files, or did you copy from some place? If they are available in some other repository, they can be download by conan, only need to add their urls to conandata.yml. It's rare, but it happens, we have some projects which download more than one file using conandata.

File: recipes/fbclient/3.x.x/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    generators = "cmake", "cmake_find_package_multi"
+    # NOTE: The cpp files are from epp files pre-processed using the gpre tool
+    #       See https://github.com/conan-io/conan-center-index/issues/9307
+    exports_sources = "CMakeLists.txt", "array.cpp", "blob.cpp"
```
Comment (friendlyanon) [inline recipes/fbclient/3.x.x/conanfile.py:32]: @uilianries The files can be found here https://github.com/FirebirdSQL/firebird/tree/v3.0.8/src/yvalve

Comment (uilianries) [reply recipes/fbclient/3.x.x/conanfile.py:32]: @friendlyanon Thank you for the clarification. In this case, you can download them directly, don't need to copy. As real example, you may check ApprovalTests conandata: https://github.com/conan-io/conan-center-index/blob/master/recipes/approvaltests.cpp/all/conandata.yml

**Review by jgsogo — COMMENTED** [review]
Two comments here:
 * IMHO this should be a recipe for `firebird` (not only the client)... on my side, it would be ok to build only the client now if building everything is too complex. 
 * Regarding https://github.com/conan-io/conan-center-index/issues/9307, it also happens for `icu` and it just works. You can build-requires the same reference (https://github.com/conan-io/conan-center-index/blob/master/recipes/icu/all/conanfile.py#L122). Using this approach you should be able to generate `gpre` and generate those files, right?
 
 Thanks!

**Pull request closed without merging**